tv_shows_app/src/components/features/shows/ShowSection.tsx

```diff
@@ -1,53 +1,90 @@
 import { Flex, Image, Text, Box } from '@chakra-ui/react';
 import { ShowDetais } from './ShowDetails';
-import { IReview } from '../reviews/ReviewItem';
+import { IReview } from '../reviews/ReviewItem/ReviewItem';
 import { useEffect, useState } from 'react';
 import { ShowReviewSection } from './ShowReviewSection';
 import { IShow } from '@/typings/show';
+import { swrKeys } from '@/fetchers/swrKeys';
+import {
+  createReviewMutator,
+  deleteReviewMutator,
+  getMutator,
+} from '@/fetchers/mutators';
+import useSWRMutation from 'swr/mutation';
+import useSWR, { mutate } from 'swr';
+import { create } from 'domain';
 
 interface IShowSection {
   show: IShow;
 }
 
+interface ICreateReviewParams {
+  comment: string;
+  rating: number;
+  show_id: number;
+}
+
+interface IListReviewsParams {
+  show_id: number;
+}
+
 export const ShowSection = ({ show }: IShowSection) => {
-  let tempList = [
-    { email: '', avatarUrl: '', rating: 3, comment: 'Dobar film :D' },
-    { email: '', avatarUrl: '', rating: 4, comment: 'Loš film >:(' },
-  ];
-  const [reviews, setReviews] = useState(tempList);
+  const [reviews, setReviews] = useState<IReview[]>([]);
   useEffect(() => {
-    const loadedList = loadFromLocalStorage();
-    setReviews(loadedList);
+    listReviews({ show_id: show.id });
   }, []);
 
-  const loadFromLocalStorage = () => {
-    console.log(show.id);
-    const lsValue = localStorage.getItem('infinum-reviews-' + show.id);
-    if (!lsValue) return tempList;
-    return JSON.parse(lsValue);
-  };
-  const saveToLocalStorage = (newList: IReview[]) => {
-    localStorage.setItem('infinum-reviews-' + show.id, JSON.stringify(newList));
-  };
-  function onAdd(review: IReview) {
-    const newList = [...reviews, review];
+  const { trigger: triggerList } = useSWRMutation(
+    swrKeys.listReviews(show.id),
+    getMutator<IListReviewsParams>,
+    {
+      onSuccess: (data) => {
+        console.log(data);
+        setReviews(data.reviews);
+      },
+    }
+  );
+  async function listReviews(params: IListReviewsParams) {
+    const data = await triggerList(params);
+    mutate(`/api/shows/${params.show_id}`);
+    return data;
+  }
```

Comment:
Ovo smo rekli zamijeniti sa `useSWR` jer radimo GET, pa nakon toga nam ne treba ni `useEffect`, ni custom state jer će to sve biti unutar `data` varijable od useSWR-a.